File: DSA/Arrays/Code/10_SortingInArrays.java
import java.util.Arrays;

// Sorting in Arrays in Java
// Bubble Sort, Selection Sort, and built-in sort

public class SortingInArrays {
    
    // Bubble Sort
    static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
    
    // Selection Sort
    static void selectionSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIdx])
                    minIdx = j;
            }
            int temp = arr[i];
            arr[i] = arr[minIdx];
            arr[minIdx] = temp;
        }
    }
    
    public static void main(String[] args) {
        int[] arr1 = {64, 34, 25, 12, 22, 11, 90};
        int[] arr2 = {64, 34, 25, 12, 22, 11, 90};
        int[] arr3 = {64, 34, 25, 12, 22, 11, 90};
        
        System.out.println("Original array: " + Arrays.toString(arr1));
        
        // Bubble Sort
        bubbleSort(arr1);
        System.out.println("After Bubble Sort: " + Arrays.toString(arr1));
        
        // Selection Sort
        selectionSort(arr2);
        System.out.println("After Selection Sort: " + Arrays.toString(arr2));
        
        // Using Java's built-in sort
        Arrays.sort(arr3);
        System.out.println("After Arrays.sort(): " + Arrays.toString(arr3));
    }
}

File: DSA/Arrays/Code/10_SortingInArrays.py
# Sorting in Arrays in Python
# Bubble Sort, Selection Sort, and built-in sort

# Bubble Sort
def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr

# Selection Sort
def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_idx = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_idx]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]
    return arr

# Insertion Sort
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key
    return arr

arr = [64, 34, 25, 12, 22, 11, 90]

print("Original array:", arr)

# Bubble Sort
arr1 = arr.copy()
bubble_sort(arr1)
print("After Bubble Sort:", arr1)

# Selection Sort
arr2 = arr.copy()
selection_sort(arr2)
print("After Selection Sort:", arr2)

# Insertion Sort
arr3 = arr.copy()
insertion_sort(arr3)
print("After Insertion Sort:", arr3)

# Using Python's built-in sort
arr4 = arr.copy()
arr4.sort()
print("Using built-in sort():", arr4)

# Using sorted() function
arr5 = sorted(arr)
print("Using sorted():", arr5)

# Sort in reverse order
arr6 = sorted(arr, reverse=True)
print("Sorted in reverse:", arr6)
